### DISCUSSION TITLE
What query is actually sent when pool_pre_ping is used with psycopg2

### DISCUSSION THREAD

**Discussion by sollymay** [sqlalchemy/sqlalchemy#11311] [Usage Questions] [answered]
I read the documentation and I understand this is dialect specific, however I am not clear on what is the actual query that is tried when `pool_pre_ping=True` when used with psycopg2?

My guess is `SELECT 1` but since this is dialect specific, wanted to know if it was different for psycopg2?

**Comment by CaselIT** [comment, marked as answer]
Hi,

psycopg2 uses `select 1`, psycopg3 uses `;` instead

**Reply by sollymay** [reply]
Sorry for the follow up but is there any way to change the default pool_pre_ping SQL statement through args/kwargs?

**Reply by CaselIT** [reply]
No, but you could likely monkey patch the dialect to change it.
There should be no particular reason to do that, though
